src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -122,16 +119,55 @@
    * @return completable future which might hold {@link InvoiceLine} on success or an exception if any issue happens
    */
   CompletableFuture<InvoiceLine> createInvoiceLine(InvoiceLine invoiceLine, Invoice invoice) {
-    invoiceLine.setId(UUID.randomUUID().toString());
+    invoiceLine.setId(UUID.randomUUID()
+      .toString());
     invoiceLine.setInvoiceId(invoice.getId());
 
     JsonObject line = mapFrom(invoiceLine);
 
-    return generateLineNumber(invoice)
-      .thenAccept(lineNumber -> line.put(INVOICE_LINE_NUMBER, lineNumber))
+    return generateLineNumber(invoice).thenAccept(lineNumber -> line.put(INVOICE_LINE_NUMBER, lineNumber))
+      .thenAccept(t -> calculateInvoiceLineTotals(invoiceLine, invoice))
```

Comment:
According to jira description:
> These totals are calculated "on the fly" and never persisted.

The calculation is done before request is sent to storage which means not readonly properties will be persisted as well

Reply:
In this case, the values are not persisted, if you look at the **createInvoiceLineSummary** method, the record in storage is being created from the "**line**" object and not the **invoiceLine** object . the calculated values are only returned in the response.